components/definitions/rewardsTip.d.ts

```diff
@@ -41,6 +41,12 @@ declare namespace RewardsTip {
     tweetText: string
   }
 
+  interface RedditMetaData {
```

Comment:
I'm wondering if we should have a more generic structure to represent media-related metadata. This is totally fine as it is since we're only supporting two sites at the moment. But I'm just thinking down the line when we keep adding media types to tip it will get unwieldy.

It might make sense to have a `MediaMetaData` type with a `category` field (Twitter, Reddit, etc.) and a dictionary of parameters. Then we could just pass the generic metadata type around and each function would know what to do with it based on the `category` it was dealing with. We did something like this on the C++ side.

Reply:
Yes, we should as more media publishers would make maintainability difficult. Planned to handle out of this PR: https://github.com/brave/brave-browser/issues/4907

Reply:
OK, great!